File: client/go/src/example/quick/card.go
package main

import (
	"fmt"
	"biostar/service/card"
)

const (
	NUM_OF_NEW_BLACKLIST = 2

	FIRST_BLACKLISTED_CARD_ID = 100000
	ISSUE_COUNT = 3
)


func testCard(deviceID uint32) error {
	fmt.Printf(">>> Scan a card...\n")

	cardData, err := cardSvc.Scan(deviceID)

	if err != nil {
		return err
	}

	fmt.Printf("Card: %v\n\n", cardData)

	blacklist, err := cardSvc.GetBlacklist(deviceID)

	if err != nil {
		return err
	}

	fmt.Printf("Blacklist: %v\n\n", blacklist)

	cardInfos := []*card.BlacklistItem{}

	for i := 0; i < NUM_OF_NEW_BLACKLIST; i++ {
		cardInfo := &card.BlacklistItem{
			CardID: []byte(fmt.Sprintf("%v", FIRST_BLACKLISTED_CARD_ID + i)),
			IssueCount: ISSUE_COUNT,
		}

		cardInfos = append(cardInfos, cardInfo)
	}

	err = cardSvc.AddBlacklist(deviceID, cardInfos)

	if err != nil {
		return err
	}

	blacklist, _ = cardSvc.GetBlacklist(deviceID)
	fmt.Printf("Blacklist after adding new items: %v\n\n", blacklist)	

	err = cardSvc.DeleteBlacklist(deviceID, cardInfos)

	if err != nil {
		return err
	}

	blacklist, _ = cardSvc.GetBlacklist(deviceID)
	fmt.Printf("Blacklist after deleting new items: %v\n\n", blacklist)	

	return nil
}

File: client/csharp/example/quick/Card.cs
using System;
using Google.Protobuf;

namespace example
{
    class CardTest
    {
        private const int NUM_OF_BLACKLIST_ITEM = 2;      
        private const int FIRST_BLACKLISTED_CARD_ID = 100000;
        private const int ISSUE_COUNT = 3;

        private CardSvc cardSvc;

        public CardTest(CardSvc svc) {
            cardSvc = svc;
        }

        public void Test(uint deviceID) {
            Console.WriteLine(">>> Scan a card...");
            
            var cardData = cardSvc.Scan(deviceID);

            Console.WriteLine("Card data: {0}" + Environment.NewLine, cardData);

            var blacklist = cardSvc.GetBlacklist(deviceID);

            Console.WriteLine("Blacklist: {0}" + Environment.NewLine, blacklist);

            var newCardInfos = new Card.BlacklistItem[NUM_OF_BLACKLIST_ITEM];

            for(int i = 0; i < NUM_OF_BLACKLIST_ITEM; i++) {
            newCardInfos[i] = new Card.BlacklistItem{ CardID = ByteString.CopyFromUtf8(String.Format("{0}", FIRST_BLACKLISTED_CARD_ID + i)), IssueCount = ISSUE_COUNT };
            }

            cardSvc.AddBlacklist(deviceID, newCardInfos);

            blacklist = cardSvc.GetBlacklist(deviceID);
            Console.WriteLine("Blacklist after adding new items: {0}" + Environment.NewLine, blacklist);

            cardSvc.DeleteBlacklist(deviceID, newCardInfos);
            blacklist = cardSvc.GetBlacklist(deviceID);
            Console.WriteLine("Blacklist after deleting new items: {0}" + Environment.NewLine, blacklist);
        }        
    }
}

